crates/bevy_ui/src/layout_components.rs

```diff
@@ -0,0 +1,430 @@
+//! Components used to control the layout of UI-[`Node`](super::Node) entities.
+use crate::{Size, UiRect, Val};
+use bevy_derive::{Deref, DerefMut};
+use bevy_ecs::prelude::Component;
+use bevy_reflect::prelude::*;
+use serde::{Deserialize, Serialize};
+
+use self::flex::FlexLayout;
+
+/// Controls which layout algorithm is used to position this UI node
+#[derive(Component, Copy, Clone, PartialEq, Debug, Serialize, Deserialize, Reflect)]
+#[reflect_value(PartialEq, Serialize, Deserialize)]
+pub enum Layout {
+    /// Use the absolute position and size specified.
+    PositionLayout,
+    /// Use the [Flexbox](https://cssreference.io/flexbox/) layout algorithm.
+    FlexLayout(FlexLayout),
```

Comment:
Do we need to repeat `Layout` in the enum value, since it's already in the enum name and as far as I know you can't use a value without prefixing it by its enum type name anyway, so it's redundant?

Reply:
You can use the value without the enum type prefix, if you `use` the enum variants ([Playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=4b7ea7ccab90f8289dae91ae70354966)).

But your point probably still stands.

Reply:
Agreed, we should cut Layout from the enum values.